lib/message_bus/backends/redis.rb

```diff
@@ -278,7 +284,7 @@ def global_subscribe(last_id = nil, &blk)
         end
 
         begin
-          @redis_global = new_redis_connection
+          @redis_global ||= new_redis_connection
```

Comment:
hmmm this though is changing behavior, if there are 2 global subscribes then they share a connection so this will get stuck. 

I guess instead if there is a redis global already we could either exception out or disconnect it, or hold an array of redis globals

Reply:
It isn't possible to have 2 global subscribers from what I can tell. Each instance of messageBus will only have one subscriber thread.

```
  def ensure_subscriber_thread
    @mutex.synchronize do
      return if (@subscriber_thread && @subscriber_thread.alive?) || @destroyed

      @subscriber_thread = new_subscriber_thread
    end
  end
```

Reply:
yeah so `||=` feels odd here?

Reply:
OOOOO but we allow people to do a blocking subscribe...

```
  def blocking_subscribe(channel = nil, &blk)
    if channel
      reliable_pub_sub.subscribe(encode_channel_name(channel), &blk)
    else
      reliable_pub_sub.global_subscribe(&blk)
    end
  end
```

Reply:
I need to do a full review of the code path here. With `blocking_subscribe`, the `@redis_global` can be reassigned and doesn't seem to be thread safe.

Reply:
yikes, I guess to be 100% correct we need to start tracking an array of `redis_global` connections so unsubscribe will clean them all up.